src/stdio_mgr/stdio_mgr.py

```diff
@@ -27,12 +27,58 @@
 """
 
 import sys
-from contextlib import contextmanager
-from io import BufferedReader, BytesIO, StringIO, TextIOBase, TextIOWrapper
+from contextlib import contextmanager, ExitStack, suppress
+from io import BufferedRandom, BufferedReader, BytesIO, TextIOBase, TextIOWrapper
 
 import attr
 
 
+class _PersistedBytesIO(BytesIO):
+    """Class to persist the stream after close.
+
+    A copy of the bytes value is available at :attr:`_closed_buf` after
+    the :meth:`~BytesIO.close`.
+    """
+
+    def close(self):
+        self._closed_buf = self.getvalue()
+        super().close()
+
+
+class RandomTextIO(TextIOWrapper):
+    """Class to capture writes to a buffer even when detached.
+
+    Subclass of :cls:`~io.TextIOWrapper` that utilises an internal
+    buffer defaulting to utf-8 encoding.
+
+    All writes are flushed to the buffer.
+
```

Comment:
Just realized: `RandomTextIO` will *also* not be thread-safe. Should probably replicate here that warning from `TeeStdin`.

Reply:
We might want to try to find why that note was added to the python docs; it may help us understand/explain if/why it isnt thread-safe, and possibly ways to make it thread-safe.

But it is safer to include the note again, until we figure that out.

I ran out of steam tonight - will finish this first thing tmr.

Reply:
Added as #41

Reply:
https://bugs.python.org/issue10478 sheds more light on it, and the sample there is definitely what someone could try to do with stdio-mgr .  While very old

> Also, the outer TextIOWrapper (we're talking about sys.stdout here) is not designed to MT-safe at all and is probably in an inconsistent state itself.

A quick scan of `textio.c` doesnt reveal much more to me about what is unsafe - no specific notes.

So, will add the note and move on.